wpgtk: init at 5.7.4

Melkor333 opened the pull request:
###### Motivation for this change

It needs `"programs.dconf.enabled = true"` in the NixOS configuration or `dconf` installed and running on other distributions. Since this can't be tested in the package, I added this information in the long description. I hope it is i.O. like that.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) _**(excluding hidden wrappers)**_
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Melkor333 committed:
wpgtk: init at 5.7.4

jtojnar commented on pkgs/tools/X11/wpgtk/default.nix:
```diff
@@ -0,0 +1,49 @@
+    sha256 = "0c0kmc18lbr7nk3hh44hai9z06lfsgwxnjdv02hpjwrxg40zh726";
+  };
+
+  pythonEnv = python3.withPackages(ps: with ps; [ pygobject3 pillow pip]);
```
You should not use both `python` and `pythonPackages` in a single package – it can lead to issues when user overrides one but not the other.

jtojnar replied:
Also, since this package uses `buildPythonApplication`, adding the libraries to `buildInputs` or `pythonPath` attribute should be enough.

jtojnar commented on pkgs/tools/X11/wpgtk/default.nix:
```diff
@@ -0,0 +1,49 @@
+  gobjectIntrospection, gtk3, wrapGAppsHook, gnome3 }:
+
+python36Packages.buildPythonApplication rec {
+  name = "${pname}-${version}";
```
This line can be removed. Packages built with `buildPythonApplication` do this by default.

jtojnar commented on pkgs/tools/X11/wpgtk/default.nix:
```diff
@@ -0,0 +1,49 @@
+    gnome3.adwaita-icon-theme
+    pythonEnv
+    pywal
+    feh
```
Adding programs to inputs will most likely not be enough. See https://stackoverflow.com/a/34837585/160386 for a description of how runtime dependencies work.

Melkor333 replied:
I added them to `buildInputs` respectively `pythonPath` and it worked when I tested it. `feh` is being used by the program on runtime and I tested that successfully (feh isn't installed in my environment). Can I assume that this is not an issue or is there a way to test it?

jtojnar replied:
Grepping the source code, `feh` is not even used.

Melkor333 commented:
Thank you very much for your good review! I have changed the package so that it should now be okay:
- fixed typos
- removed `python3` and use only python36Packages now
- removed pythonEnv and added necessary libraries to pythonPath.
Thank you very much for pointing this out. I initially only used buildInputs which didn't work and found propagatedBuildInputs being used in another package. Both work but pythonPath is obviously the proper variable to use
- remove unnecessary `name` variable

jtojnar commented on pkgs/tools/X11/wpgtk/default.nix:
```diff
@@ -0,0 +1,50 @@
+  pythonPath = [
+    python36Packages.pygobject3
+    python36Packages.pillow
+    python36Packages.pip
```
Why pip?

jtojnar commented:
Could you try rebasing to fix the conflict and squashing to a single commit?

Melkor333 commented:
Finally did it. Sorry for this delay and thanks again very much for your time.

jtojnar commented:
Thank you.
